### PR TITLE
Fix for #142. Use the cwd if no path is specified.

### PR DESCRIPTION
**Pull request by paulhuggett**

### CODE REVIEW

File: llvm/lib/MC/MCRepoTicketFile.cpp [hunk 1]
```diff
@@ -319,11 +319,20 @@ llvm::mc::repo::realTicketDirectory(const StringRef &Path) {
...
+  llvm::SmallString<256> Result;
+  if (Out.size() == 0) {
+    // No path was specified: just use the cwd.
+    if (const std::error_code EC = llvm::sys::fs::current_path(Result)) {
+      return llvm::errorCodeToError(EC);
+    }
+    return Result;
+  }
+
   llvm::SmallString<256> Out2;
```
Comment (MaggieYingYi) [inline llvm/lib/MC/MCRepoTicketFile.cpp:331]: The variable Out2 is not used, should it be removed?

**Review by MaggieYingYi — COMMENTED** [review]
Is it possible to add a lit test for this issue?

File: llvm/lib/MC/MCRepoTicketFile.cpp (hunk 1 above)
Comment (paulhuggett) [reply llvm/lib/MC/MCRepoTicketFile.cpp]: Yes. I've simplified the change.

**Comment by paulhuggett** [comment]
> Is it possible to add a lit test for this issue?

I don’t know. Do its various shells all support changing the working directory as part of the test?

**Comment by MaggieYingYi** [comment]
> 
> 
> > Is it possible to add a lit test for this issue?
> 
> I don’t know. Do its various shells all support changing the working directory as part of the test?

The test is for the repo-create-ticket tool. The output doesn't support only filename (without path) before you fix the issue. Your change make the --output support the filename. We could add a test for the repo-create-ticket, just like we did for repo2obj tool.

**Comment by paulhuggett** [comment]
> The test is for the repo-create-ticket tool. The output doesn't support only filename (without path) before you fix the issue. Your change make the --output support the filename. We could add a test for the repo-create-ticket, just like we did for repo2obj tool.

Yes, but that doesn’t answer my question! I don’t know if it’s possible to write such a test. There’s no equivalent test for repo2obj.

**Comment by MaggieYingYi** [comment]
> 
> Yes, but that doesn’t answer my question! I don’t know if it’s possible to write such a test. There’s no equivalent test for repo2obj.
> 

Sorry, I didn't quite get what you said. 

I think we can add a test as you mentioned in the issue https://github.com/SNSystems/llvm-project-prepo/issues/142#issue-854353263

~~~
% repo-create-ticket -o foo.o --repo=clang.db 51fdad67cdc0518e6113954e60de80d9
error: No such file or directory
~~~

Before the fix, the test failed. After this fix, the test passed.

Does it make sense to you?

**Comment by paulhuggett** [comment]
> I think we can add a test as you mentioned in the issue [#142 (comment)](https://github.com/SNSystems/llvm-project-prepo/issues/142#issue-854353263)
> 
> ```
> % repo-create-ticket -o foo.o --repo=clang.db 51fdad67cdc0518e6113954e60de80d9
> error: No such file or directory

Unfortunately, this will write a file ”foo.o” into a random directory (which ever happens to be the working directory when the test ran). I don’t think that’s acceptable. For the test to be possible, I would need to set the cwd to a known location.

**Comment by MaggieYingYi** [comment]
> 
> 
> > I think we can add a test as you mentioned in the issue [#142 (comment)](https://github.com/SNSystems/llvm-project-prepo/issues/142#issue-854353263)
> > ```
> > % repo-create-ticket -o foo.o --repo=clang.db 51fdad67cdc0518e6113954e60de80d9
> > error: No such file or directory
> > ```
> 
> Unfortunately, this will write a file ”foo.o” into a random directory (which ever happens to be the working directory when the test ran). I don’t think that’s acceptable. For the test to be possible, I would need to set the cwd to a known location.

Thanks for explanation. I see now.

**Review by MaggieYingYi — APPROVED** [review]
LGTM!

**Pull request merged**